modules/drivers/redshift/test/metabase/driver/redshift_test.clj

```diff
@@ -400,76 +400,87 @@
+        (let [schema-name                  (redshift.test/unique-session-schema)
+              username                     (str (sql.tu.unique-prefix/unique-prefix) "privilege_rows_test_role")
+              db-name                      (:name database)
+              table-name                   (tx/db-qualified-table-name db-name "table")
+              qual-tbl-name                (format "\"%s\".\"%s\"" schema-name table-name)
+              table-partial-select-name    (tx/db-qualified-table-name db-name "tbl_sel")
+              qual-tbl-partial-select-name (format "\"%s\".\"%s\"" schema-name table-partial-select-name)
+              table-partial-update-name    (tx/db-qualified-table-name db-name "tbl_upd")
+              qual-tbl-partial-update-name (format "\"%s\".\"%s\"" schema-name table-partial-update-name)
+              view-nm                      (tx/db-qualified-table-name db-name "view")
+              qual-view-name               (format "\"%s\".\"%s\"" schema-name view-nm)
+              mview-name                   (tx/db-qualified-table-name db-name "mview")
+              qual-mview-name              (format "\"%s\".\"%s\"" schema-name mview-name)
+              conn-spec                    (sql-jdbc.conn/db->pooled-connection-spec database)
+              get-privileges               (fn []
+                                             (sql-jdbc.conn/with-connection-spec-for-testing-connection
+                                               [spec [:redshift (assoc (:details database) :user username)]]
+                                               (with-redefs [sql-jdbc.conn/db->pooled-connection-spec (fn [_] spec)]
+                                                 (set (sql-jdbc.sync/current-user-table-privileges driver/*driver* spec)))))]
           (try
            (execute! (format
                       (str
                        "CREATE TABLE %1$s (id INTEGER);\n"
                        "CREATE VIEW %2$s AS SELECT * from %1$s;\n"
                        "CREATE MATERIALIZED VIEW %3$s AS SELECT * from %1$s;\n"
-                       "CREATE USER \"%4$s\" WITH PASSWORD '%5$s';\n"
-                       "GRANT SELECT ON %1$s TO \"%4$s\";\n"
-                       "GRANT UPDATE ON %1$s TO \"%4$s\";\n"
-                       "GRANT SELECT ON %2$s TO \"%4$s\";\n"
-                       "GRANT SELECT ON %3$s TO \"%4$s\";")
+                       "CREATE TABLE %4$s (id INTEGER);\n"
+                       "CREATE TABLE %5$s (id INTEGER);\n"
+                       "CREATE USER \"%6$s\" WITH PASSWORD '%7$s';\n"
+                       "GRANT SELECT ON %1$s TO \"%6$s\";\n"
+                       "GRANT SELECT (id) ON %4$s TO \"%6$s\";\n"
+                       "GRANT UPDATE (id) ON %5$s TO \"%6$s\";\n"
+                       "GRANT UPDATE ON %1$s TO \"%6$s\";\n"
+                       "GRANT SELECT ON %2$s TO \"%6$s\";\n"
+                       "GRANT SELECT ON %3$s TO \"%6$s\";")
```

Comment:
having the grant statements in the same order as their corresponding args makes this easier to read imo
```suggestion
                       "GRANT UPDATE ON %1$s TO \"%6$s\";\n"
                       "GRANT SELECT ON %2$s TO \"%6$s\";\n"
                       "GRANT SELECT ON %3$s TO \"%6$s\";\n"
                       "GRANT SELECT (id) ON %4$s TO \"%6$s\";\n"
                       "GRANT UPDATE (id) ON %5$s TO \"%6$s\";")
```